src/parser.ts

```diff
@@ -553,12 +555,25 @@ function assignOptionalAndDefaultToName({
     }
   }
 
+  if ([DEFAULT, DEFAULT_Value].includes(tag)) {
+    const emptyArrayOrObjectRegEx = /(\[ *])|({ *}) *$/
+    const usefulSourceLine = source.find(x => x.source.includes(`@${tag}`))?.source || ''
+    const matchResult = usefulSourceLine.match(emptyArrayOrObjectRegEx)
+
+    if (!name && matchResult) {
+      const { 1: array, 2: object } = matchResult
+      // The space is to improve readability in non-monospace fonts
+      name = (array && '[ ]') || (object && '{ }') || ''
```

Comment:
Why you don't added to `type` instead of name

Reply:
When I apply it to type, it doesn't produce the correct results.

Here are some examples from failed tests (top value being expected and bottom value being reality):

![image](https://user-images.githubusercontent.com/10610368/162560028-baf24e66-38ba-447b-a13c-4378dbe318ef.png)

![image](https://user-images.githubusercontent.com/10610368/162560062-796dea85-2f22-4d54-8fb9-375a233d0028.png)

Reply:
We should handle it here https://github.com/hosseinmd/prettier-plugin-jsdoc/blob/cb91a5cae01927f2e935aea82b78f4ec09412300/src/stringify.ts#L67